src/oc_erchef/apps/oc_chef_wm/src/oc_chef_data_collector.erl

```diff
@@ -0,0 +1,60 @@
+-module(oc_chef_data_collector).
+
+-include("oc_chef_wm.hrl").
+
+-export([notify/2]).
+
+-ifdef(TEST).
+-compile([export_all]).
+-endif.
+
+-spec notify(Req :: wm_req(), State :: #base_state{}) -> ok.
+notify(Req, #base_state{resource_state = ResourceState, resource_mod = ResourceMod} = State) when
+    is_record(ResourceState, acl_state);
+    is_record(ResourceState, association_state);
+    is_record(ResourceState, client_state);
+    %% Skip controls because oc_chef_action can't extract them yet
+    %% is_record(ResourceState, control_state);
+    is_record(ResourceState, cookbook_state);
+    is_record(ResourceState, data_state);
+    is_record(ResourceState, environment_state);
+    is_record(ResourceState, group_state);
+    is_record(ResourceState, node_state);
+    is_record(ResourceState, organization_state);
+    is_record(ResourceState, role_state);
+    is_record(ResourceState, user_state);
+    is_record(ResourceState, key_state) ->
+        case {wrq:method(Req), req_success(Req), ResourceMod} of
+            %% Skip GET because we're not updating anything
+            {'GET', _, _} ->
+                ok;
+            %% authenticate_user should not be an action.
+            {_, _, oc_chef_wm_authenticate_user} ->
+                ok;
+            %% Notify any other 2XX message for matching resources
+            {_, true, _} ->
+                Msg = oc_chef_action:create_message(Req, State, true),
+                data_collector_http:post("/", Msg);
```

Comment:
I'm I'm reading the control flow correctly, this is going to happen synchronously as part of the request.  Given that, I think we should consider some of the following:

- Logging errors that occur here rather than silently failing.
- Wrapping this call, at some level, in a stats_hero timer so that we generate statsd metrics and request log annotations with timing information about how long the post has taken.  Here is an example of how you might do that: https://github.com/chef/chef-server/blob/master/src/oc_erchef/apps/chef_index/src/chef_index.erl#L142-L144